### DISCUSSION TITLE
Isn't this basically web components?

### DISCUSSION THREAD

**Discussion by emilio-martinez** [blockprotocol/blockprotocol#149] [Q&A] [answered]
I ran into this project, and I genuinely wonder whether it has been considered that this is basically what [Web Components](https://developer.mozilla.org/en-US/docs/Web/Web_Components) do in the web today:

- They are already a web standard
- They are framework agnostic
- And they can be built with any framework, including React, if desired
- They have a registry
- They can be encapsulated

Am I missing anything? If so, wouldn't these efforts better be directed towards building on top of the existing standards?

Thanks!

P.S.: The website literally says "Blocks are web components" which may be misleading if this isn't actually referring to the actual standard for Web Components.

**Comment by CiaranMn** [comment, marked as answer]
One additional thing that the Block Protocol provides which Web Components do not is standardising the interface between the component and the embedding application - e.g. the methods or events the WC is using. 

The Block Protocol specifies the name and purpose of operations for applications to provide to blocks, and blocks to use to do things. Web Components can _describe_ their interface programmatically, but each one can be different - they are described but not standardized. This means that you often need to know the API of a specific WC in order to implement it. The aim of the Block Protocol is that all blocks have the same API, and thus new blocks can be added to an application without having to learn anything about them.

Web Components are one way of implementing such a block - we mention this briefly [here](https://blockprotocol.org/spec/implementation-approaches#web-components), although it needs fleshing out and adding to the FAQ, since it is one!

We are indeed aiming to build on top of the existing standards: we don't want to invent a new way of implementing encapsulated web components (there's already several of those), but instead standardize how those encapsulated components communicate with the application using them. There's a bit more in the [FAQ](https://blockprotocol.org/docs/faq) about our relation to existing standards.

**Reply by vilkinsons** [reply]
Another version of this answer can be found in our [FAQ](https://blockprotocol.org/docs/faq#how-are-block-protocol-blocks-different-to-web-components).
